spinalcordtoolbox/reports/slice.py

```diff
@@ -340,6 +340,8 @@ def _resample_slicewise(self, image, p_resample, type_img, image_ref=None):
         dict_interp = {'im': 'spline', 'seg': 'linear'}
         # Create nibabel object
         nii = Nifti1Image(image.data, image.hdr.get_best_affine())
+        # Check if input image is binary
+        is_binary = np.isin(nii.get_data(), [0, 1]).all()
```

Comment:
Hmm, will this work if the image is float? I'm not sure off the top of my head how `np.isin` works for float/int comparison.

Reply:
Also, one other small small nitpick -- Do we need to do this check here?

If `is_binary` is only used in a single place, maybe we could just have this directly inside the if/else block.

Reply:
Eh, honestly, this is too much of a nitpick to be worth fixing. Scratch my recommendation. :P